spec/rubocop/cop/rspec/duplicated_metadata_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::DuplicatedMetadata do
+  context 'when metadata is not used' do
+    it 'registers no offense' do
+      expect_no_offenses(<<~RUBY)
+        describe 'Something' do
+        end
+      RUBY
+    end
+  end
+
+  context 'when metadata is not duplicated' do
+    it 'registers no offense' do
+      expect_no_offenses(<<~RUBY)
+        describe 'Something', :a, :b do
+        end
+      RUBY
+    end
+  end
+
+  context 'when metadata is duplicated on example group' do
+    it 'registers offense' do
+      expect_offense(<<~RUBY)
+        describe 'Something', :a, :a do
+                                  ^^ Avoid duplicated metadata.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        describe 'Something', :a do
+        end
+      RUBY
+    end
+  end
+
+  context 'when metadata is duplicated in different order' do
+    it 'registers offense' do
+      expect_offense(<<~RUBY)
+        describe 'Something', :a, :b, :a do
+                                      ^^ Avoid duplicated metadata.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        describe 'Something', :a, :b do
+        end
+      RUBY
+    end
+  end
+
+  context 'when metadata is duplicated on example' do
+    it 'registers offense' do
+      expect_offense(<<~RUBY)
+        it 'does something', :a, :a do
+                                 ^^ Avoid duplicated metadata.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        it 'does something', :a do
+        end
+      RUBY
+    end
+  end
+
+  context 'when metadata is duplicated on shared examples' do
+    it 'registers offense' do
+      expect_offense(<<~RUBY)
+        shared_examples 'something', :a, :a do
+                                         ^^ Avoid duplicated metadata.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        shared_examples 'something', :a do
+        end
+      RUBY
+    end
+  end
+
+  context 'when metadata is duplicated on configuration hook' do
+    it 'registers offense' do
+      expect_offense(<<~RUBY)
+        RSpec.configure do |configuration|
+          configuration.before(:each, :a, :a) do
+                                          ^^ Avoid duplicated metadata.
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        RSpec.configure do |configuration|
+          configuration.before(:each, :a) do
```

Comment:
Is `:each` the default? 
Would it still work? https://github.com/rspec/rspec-core/blob/6dede8ae1b9cce6e1de4baa912e77c97b587b446/lib/rspec/core/hooks.rb#L584
```ruby
configuration.before(:a) do
```

`#Hooks.all _ ${send` expects a scope, right?

Reply:
You are right that the default scope is `:each` (`:example` to be exact), but the code `configuration.before(:a)` is not allowed because you must explicitly pass the scope as a first argument when passing metadata by Symbol to hook methods.

- https://github.com/rspec/rspec-core/blob/6dede8ae1b9cce6e1de4baa912e77c97b587b446/lib/rspec/core/hooks.rb#L591-L603

Reply:
Fair enough 👍 
What about hash metadata, e.g.:
```ruby
configuration.before(a: true, a: true) do
```
Probably not a big deal, as another cop would tell about a duplicate hash element, and even if not, Ruby would collapse it.

Reply:
I was going to make a point that this not only relates to this cop, but I've realized that this was a previously existing code, so I take this argument back.